src/main/java/uk/gov/hmcts/dm/service/BlobStorageDeleteService.java

```diff
@@ -78,4 +61,41 @@ public void deleteDocumentContentVersion(@NotNull DocumentContentVersion documen
         }
     }
 
+    private boolean deleteDocumentBinary(DocumentContentVersion documentContentVersion) {
+        BlockBlobClient blob =
+            cloudBlobContainer.getBlobClient(documentContentVersion.getId().toString()).getBlockBlobClient();
+        if (TRUE.equals(blob.exists())) {
+            Response<Void> res = blob.deleteWithResponse(
+                DeleteSnapshotsOptionType.INCLUDE, null, null, null);
+            if (res.getStatusCode() != 202 && res.getStatusCode() != 404) {
+                log.info(
+                    "Deleting document blob {} failed. Response status code {}",
+                    documentContentVersion.getId(),
+                    res.getStatusCode()
+                );
+                return false;
+            }
+            log.info(
+                "Successfully deleted blob: {}, document {}, StoredDocument {}",
+                blob.getBlobUrl(),
+                documentContentVersion.getId(),
+                documentContentVersion.getStoredDocument().getId()
+            );
+        }
+        return true;
+    }
+
+    public void deleteCaseDocumentBinary(DocumentContentVersion documentContentVersion) {
+        try {
+            if (!deleteDocumentBinary(documentContentVersion)) {
+                throw new Exception("Blob could not be deleted");
```

Comment:
do we need to throw?
we log similar things in other function

Reply:
Yes. Throwing a Exception made more sense here.